src/pages/ExplorerPage/ExplorerPage.tsx

```diff
@@ -35,42 +36,87 @@ const ExplorerPage = () => {
 
   const classes = useStyles();
   const [active, setActive] = useState(false);
+  // NOTE: we are going to use the setter upon filters implementation
   const [queryParams, setQueryParams] = useState({ page: "1", limit: "10" });
 
-  const [state, setState] = useState<{ transfers: Transfer[] | never[], loading: "none" | "loading" | "done", isReady: boolean }>({
+  const [state, setState] = useState<{
+    transfers: Transfer[] | never[];
+    loading: "none" | "loading" | "done";
+    isReady: boolean;
+    error: undefined | string;
+  }>({
     transfers: [],
     loading: "none",
-    isReady: false
-  })
+    isReady: false,
+    error: undefined,
+  });
 
   const handleTimelineButtonClick = () =>
     explorerPageDispatcher({ type: "timelineButtonClick" });
 
   const transferData = async () => {
-    const transfersResponse = await routes.transfers(
-      queryParams.page,
-      queryParams.limit,
-    );
-    setState((prevState) => ({
-      ...prevState,
-      transfers: transfersResponse,
-      loading: "loading",
-      isReady: true
-    }))
+    try {
+      const transfersResponse = await routes.transfers(
+        queryParams.page,
+        queryParams.limit,
+      );
+      setState((prevState) => ({
+        ...prevState,
+        transfers: transfersResponse,
+        loading: "loading",
+        isReady: true,
+      }));
+    } catch (e) {
+      setState((prevState) => ({
+        ...prevState,
+        error: "Error fetching all the transfers",
+      }));
+    }
   };
 
   useEffect(() => {
-    transferData();
+    if (explorerState.account === undefined) {
+      transferData();
+    }
   }, []);
 
+  const transferDataBySender = async (sender: string) => {
+    try {
+      const transferResponseBySender = await routes.transferBySender(
+        sender,
+        "1",
+        "10",
+      );
```

Comment:
These should be using the state values for `page` and `limit` if I understand things correctly.

Reply:
since we don't have paginaton, that is a separate issue that will be there after with address that. I'm going to change those values to come from directly from state initialization though 😄

Reply:
done